### PR TITLE
Dynamic timefilter

### PR DESCRIPTION
**Pull request by rbdebeer**
I have added more dynamic date/time searching. 

- The "start" time range input box supports date/time formatting with ranges. Ranges supported are [+|-|+-][y/m/d/h/m/s]. Eg. you can now search for 2017-08-10 +-3d to search 3 days before and after this data. 2017-08-10 14:21 +5m would search 5 minutes after that date/time.
- The data/time cells in the datagrid are left clickable. A left click will add the date to the input box of the time range search.

### CODE REVIEW

File: timesketch/ui/static/components/explore/explore-filter-directive.js [hunk 1]
```diff
@@ -68,8 +70,37 @@ limitations under the License.
...
+                scope.parseFilterDate = function(datevalue){  
+                    if (datevalue != null) {
+                        var datetimetemplate="YYYY-MM-DDTHH:mm:ss";
+                        //Parse out 'T' date time seperator needed by ELK but not by moment.js
+                        datevalue=datevalue.replace(/T/g,' ');
+                        //Parse offset given by user. Eg. +-10m
+                        var offsetRegexp = /(.*?)(-|\+|\+-|-\+)(\d+)(y|d|h|m|s|M|Q|w|ms)/g;
+                        var match = offsetRegexp.exec(datevalue);
+                        if (match != null) {
+                            match[1] = moment(match[1],"YYYY-MM-DD HH:mm:ssZZ");
```
Comment (berggren) [inline timesketch/ui/static/components/explore/explore-filter-directive.js]: It's a bit hard to follow here without looking at the regexp. Could you put the mathed expressions in variables perhaps? Just to make it easier to read.

**Comment by rbdebeer** [comment]
Thanks. Added regexp matches in variables for code readability.

**Comment by berggren** [comment]
Ref issue: #1
